### PR TITLE
Parsable object layers

### LINKED ISSUE #24: ObjectLayers should be parsable when multiple exist
#18 Needs to work when multiple ObjectLayers are present, these will represent multiple Guards.

### PR DESCRIPTION
**Pull request by ForbiddenKJ**
all objects when being parsed are sorted into their types and returned in a dictionary by these sorted types. Our return will be in the format of this `dict[str, list[ObjectLayer]]` and will look like this `{"guard": [], "key": []}`. We don't really care about the order of the elements in the array, this is fine although the types are relevant.

Closes #24

### CODE REVIEW

**Review by GDWR — APPROVED** [review]
Looks good, would be better if we could have a more abstract method to allow for other object layers in the future. But time constraints are a thing.

File: __main__.py [hunk 1]
```diff
@@ -67,7 +67,15 @@ def load_map(self):
 
         self.object_layers = utils.process_objects(f"assets/tilemaps/TestLevel.tmx")
 
-        self.guard1_locations = utils.extract_guard_locations(self.object_layers[0])
+        self.guard_locations = [
+            utils.extract_locations(self.object_layers["guard"][i])
+            for i in range(len(self.object_layers["guard"]))
+        ]
```
Comment (SavagePastaMan) [inline __main__.py:73]: Both these loops can be done more pythonically
```suggestion
        self.guard_locations = [
            utils.extract_locations(guard)
            for guard in self.object_layers["guard"])
        ]
```
not sure if guard is the best name, but something like this

**Review by SavagePastaMan — CHANGES_REQUESTED** [review]
looks good other than my one comment

**Comment by SavagePastaMan** [comment]
watch out for merge conflicts in `constants.py`

**Review by SavagePastaMan — APPROVED** [review]

**Pull request merged**